query/src/datasources/common/dal_metrics.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::SeekFrom;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::Poll;
+
+use common_dal::AsyncSeekableReader;
+use common_dal::DataAccessor;
+use common_dal::InputStream;
+use common_metrics::label_counter_with_val;
+use futures::Stream;
+
+pub const METRIC_DAL_READ_BYTES: &str = "dal.read.bytes";
+pub const METRIC_DAL_WRITE_BYTES: &str = "dal.write.bytes";
+
+struct InputStreamWrapper {
+    tenant_id: String,
+    cluster_id: String,
+    inner: InputStream,
+}
+
+impl InputStreamWrapper {
+    pub fn new(
+        tenant_id: impl Into<String>,
+        cluster_id: impl Into<String>,
+        input_stream: InputStream,
+    ) -> Self {
+        Self {
+            tenant_id: tenant_id.into(),
+            cluster_id: cluster_id.into(),
+            inner: input_stream,
+        }
+    }
+}
+
+pub struct DalWrapper {
+    tenant_id: String,
```

Comment:
Seems we can introduce a new struct to `common/metrics/src/recorder.rs` 
```pub struct TenantLabel {
    pub tenant_id:String,
    pub cluster_id:String,
}
```

And here we can use it as:
```
pub struct DalWrapper {
tenant_label: TenantLaber
...
}
```

Reply:
right!